src/__tests__/consumer/node_redis.spec.js

```diff
@@ -146,6 +147,183 @@ tape('NodeJS Redis', function (t) {
+          permissions: ['not_matching']
+        }), {
+          treatment: 'on',
+          config: null
+        }, 'Evaluations using Redis storage should be correct, including configs.');
+        assert.deepEqual(await client.getTreatmentWithConfig('UT_Segment_member', 'always-o.n-with-config'), {
+          treatment: 'o.n',
+          config: expectedConfig
+        }, 'Evaluations using Redis storage should be correct, including configs.');
+
+        assert.equal(await client.getTreatment('UT_Segment_member', 'always-on'), 'on', 'Evaluations using Redis storage should be correct.');
+ 
+        // Below splits were added manually to the redis_mock.json file.
+        // They are all_keys (always evaluate to on) which depend from always-on split. the _on/off is what treatment they are expecting there.
+        assert.equal(await client.getTreatment('UT_Segment_member', 'hierarchical_splits_testing_on'), 'on', 'Evaluations using Redis storage should be correct.');
+        assert.equal(await client.getTreatment('UT_Segment_member', 'hierarchical_splits_testing_off'), 'off', 'Evaluations using Redis storage should be correct.');
+        assert.equal(await client.getTreatment('UT_Segment_member', 'hierarchical_splits_testing_on_negated'), 'off', 'Evaluations using Redis storage should be correct.');
+
+        assert.equal(typeof client.track('nicolas@split.io', 'user', 'test.redis.event', 18).then, 'function', 'Track calls should always return a promise on Redis mode.');
+        assert.equal(typeof client.track().then, 'function', 'Track calls should always return a promise on Redis mode, even when parameters are incorrect.');
+
+        assert.true(await client.track('nicolas@split.io', 'user', 'test.redis.event', 18), 'If the event was succesfully queued the promise will resolve to true');
+        assert.false(await client.track(), 'If the event was NOT succesfully queued the promise will resolve to false');
+
+        await client.ready(); // promise already resolved
+        await client.destroy();
+
+        // Validate stored impressions and events
+        exec(`echo "LLEN ${config.storage.prefix}.SPLITIO.impressions \n LLEN ${config.storage.prefix}.SPLITIO.events" | redis-cli  -p ${redisPort}`, (error, stdout) => {
+          if (error) assert.fail('Redis server should be reachable');
+
+          const trackedImpressionsAndEvents = stdout.split('\n').filter(line => line !== '').map(line => parseInt(line));
+          assert.deepEqual(trackedImpressionsAndEvents, [0, 2], 'Tracked impressions and events should not be stored in Redis');
+
+          // Validate stored telemetry
+          exec(`echo "HLEN ${config.storage.prefix}.SPLITIO.telemetry.latencies \n HLEN ${config.storage.prefix}.SPLITIO.telemetry.exceptions \n HGET ${config.storage.prefix}.SPLITIO.telemetry.init nodejs-${version}/${HOSTNAME_VALUE}/${IP_VALUE}" | redis-cli  -p ${redisPort}`, (error, stdout) => {
```

Comment:
We should check that the Redis value `${config.storage.prefix}.SPLITIO.uniquekeys` is the expected